src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs

```diff
@@ -435,7 +466,8 @@ int RecursiveMatch(int firstActualIndex, int firstExpectedIndex, int unmatchedAc
                     }
 
                     var (lineSpan, additionalLineSpans) = actualResultLocations[firstActualIndex];
-                    if (!IsMatch(actualResults[firstActualIndex], actualIds[firstActualIndex], lineSpan, additionalLineSpans, actualArguments[firstActualIndex], expectedResults[i], expectedArguments[i]))
+                    var matchValue = GetMatchValue(actualResults[firstActualIndex], actualIds[firstActualIndex], lineSpan, additionalLineSpans, actualArguments[firstActualIndex], expectedResults[i], expectedArguments[i]);
+                    if (matchValue == 1)
```

Comment:
convice me that this comparison of int to double will work 100% of the time (or ad a comment about why its ok)

Reply:
As I recall mantissas etc, I remember that `0` and `1` are always safe to compare to int. So this is fine, but perhaps add a comment?

---
In reply to: [379929020](https://github.com/dotnet/roslyn-sdk/pull/455#discussion_r379929020) [](ancestors = 379929020)

Reply:
I introduced a new `MatchQuality` structure to avoid relying on `int`/`double`. I wouldn't say it's a perfect type now, but it should be much better for future refactoring.